kbchat/kbchat_test.go

```diff
@@ -103,82 +103,290 @@ func deleteWorkingDir(workingDir string) error {
 	return os.RemoveAll(workingDir)
 }
 
-var kbc *API
-var config Config
-
-func TestMain(m *testing.M) {
+func testSetup() (alice *API, config botConfig, dir string, oneOnOneChannel Channel, teamChannel Channel) {
 	var err error
 	config, err = readAndParseConfig()
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Error in reading config: %v\n", err)
 	}
-	dir, err := randomTempDir()
+	dir, err = randomTempDir()
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Error in generating directory: %v\n", err)
 	}
 	kbLocation, err := prepWorkingDir(dir)
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Error in preparing working directory: %v\n", err)
 	}
-	kbc, err = Start(RunOptions{KeybaseLocation: kbLocation, HomeDir: dir, Oneshot: config.Bots.Alice1, StartService: true})
+	alice, err = Start(RunOptions{KeybaseLocation: kbLocation, HomeDir: dir, Oneshot: config.Bots["alice1"], StartService: true})
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Error in starting service: %v\n", err)
 	}
 
-	flag.Parse()
-	code := m.Run()
+	oneOnOneChannel = Channel{
+		Name: fmt.Sprintf("%s,%s", config.Bots["alice1"].Username, config.Bots["charlie1"].Username),
+	}
+	teamChannel = Channel{
+		Name:        config.Teams["acme"].Teamname,
+		Public:      false,
+		MembersType: "team",
+		TopicName:   config.Teams["acme"].Channel,
+		TopicType:   "chat",
+	}
 
-	err = kbc.Shutdown()
+	return alice, config, dir, oneOnOneChannel, teamChannel
+}
+
+func testTeardown(alice *API, dir string) {
+	err := alice.Shutdown()
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Error on service shutdown: %v\n", err)
 	}
 	err = deleteWorkingDir(dir)
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Error working directory teardown: %v\n", err)
 	}
-	os.Exit(code)
 }
 
 func TestGetUsername(t *testing.T) {
-	require.Equal(t, kbc.GetUsername(), config.Bots.Alice1.Username)
+	alice, config, dir, _, _ := testSetup()
+	require.Equal(t, alice.GetUsername(), config.Bots["alice1"].Username)
+	testTeardown(alice, dir)
 }
 
 func TestGetConversations(t *testing.T) {
-	conversations, err := kbc.GetConversations(false)
+	alice, _, dir, _, _ := testSetup()
+	conversations, err := alice.GetConversations(false)
 	require.NoError(t, err)
 	require.Greater(t, len(conversations), 0)
+	testTeardown(alice, dir)
 }
 
 func TestGetTextMessages(t *testing.T) {
-	channel := Channel{
-		Name: fmt.Sprintf("%s,%s", config.Bots.Alice1.Username, config.Bots.Charlie1.Username),
-	}
-	messages, err := kbc.GetTextMessages(channel, false)
+	alice, _, dir, oneOnOneChannel, _ := testSetup()
+	messages, err := alice.GetTextMessages(oneOnOneChannel, false)
 	require.NoError(t, err)
 	require.Greater(t, len(messages), 0)
+	testTeardown(alice, dir)
+}
+
+func TestSendMessage(t *testing.T) {
+	alice, _, dir, oneOnOneChannel, _ := testSetup()
+	text := "test SendMessage"
```

Comment:
if it's not too hard, try to add some random text here. this will guarantee that a silent error on the `Send` doesn't unintentionally pass on the `Get` because the previous message also came from this test.